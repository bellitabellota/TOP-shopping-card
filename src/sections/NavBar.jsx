import styles from "./NavBar.module.css"
import { Link } from "react-router-dom";

function NavBar({selectedProducts}) {
  const calculateQuantityCartItems = () => {
    return selectedProducts.reduce((quantityCardItems, currentCartItem) => {
      return quantityCardItems += currentCartItem.quantity;
    }, 0);
  }

  return(
    <nav>
      <p className={styles.siteName}>Live In Style</p>
      <Link to="cart" className={styles.cardIconContainer}>
        <div className={styles.bubble}>{calculateQuantityCartItems()}</div>
        <img src="public/shopping-bag-white-icon.png" alt="cart-icon" />
        <p>Cart</p>
      </Link>
    </nav>
  )
}

export default NavBar;