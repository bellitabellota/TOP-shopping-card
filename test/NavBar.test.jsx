import { render } from "@testing-library/react";
import NavBar from "../src/sections/NavBar";
import { describe, it, expect, vi } from "vitest";
import { MemoryRouter } from "react-router-dom";

vi.mock("../src/helpers/calculateQuantityCartItems", () => ({
  default: vi.fn(() => 0 ),
}));

describe("NavBar", ()=>{
  it("renders the Navbar with an empty selectedProducts array and a mocked cart quantity of 0", () => { 
   const noSelectedItemsMock = [];
    const {container} = render(
    <MemoryRouter>
      <NavBar selectedProducts={noSelectedItemsMock} />
    </ MemoryRouter>
    )

    expect(container).toMatchSnapshot();
  })
})