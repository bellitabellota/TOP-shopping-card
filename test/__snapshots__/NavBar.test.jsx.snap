// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar > renders the Navbar with an empty selectedProducts array and a mocked cart quantity of 0 1`] = `
<div>
  <nav>
    <p
      class="_siteName_df5b48"
    >
      Live In Style
    </p>
    <a
      class="_cardIconContainer_df5b48"
      data-discover="true"
      href="/cart"
    >
      <div
        class="_bubble_df5b48"
      >
        0
      </div>
      <img
        alt="cart-icon"
        src="shopping-bag-white-icon.png"
      />
      <p>
        Cart
      </p>
    </a>
  </nav>
</div>
`;
